server.ts

```diff
@@ -2,61 +2,68 @@ import { createSecureServer } from 'http2';
 import { createServer } from 'https';
 import { route, parseArguments } from './ts-scripts/utils';
 import { readFileSync } from 'fs';
-const ip = require('my-local-ip')();
+import { serialize, parse as parserCookie } from 'cookie';
+import { compile } from 'handlebars';
+import { parse } from 'url';
+import { TBuild, TConnection, TPlatform } from './ts-scripts/interface';
+import { readFile } from 'fs-extra';
+import { join } from 'path';
 
+const ip = require('my-local-ip')();
 
-const connectionTypes = ['mainnet', 'testnet'];
-const buildTypes = ['dev', 'normal', 'min'];
 
-const privateKey = readFileSync('privatekey.pem').toString();
-const certificate = readFileSync('certificate.pem').toString();
+const connectionTypes: Array<TConnection> = ['mainnet', 'testnet'];
+const buildTypes: Array<TBuild> = ['dev', 'normal', 'min'];
 
-function createMyServer(port) {
+const privateKey = readFileSync('localhost.key').toString();
+const certificate = readFileSync('localhost.crt').toString();
 
-    const connectionTypesHash = arrToHash(connectionTypes);
-    const buildTypesHash = arrToHash(buildTypes);
+const handler = function (req, res) {
+    const url = parse(req.url);
 
-    const handler = function (req, res) {
-        const parsed = parseDomain(req.headers[':authority']);
-        if (!parsed) {
-            res.writeHead(302, { Location: `https://testnet.dev.localhost:${port}` });
-            res.end();
-        } else {
-            route(parsed.connectionType, parsed.buildType)(req, res);
-        }
-    };
+    if (url.href.includes('/choose/')) {
+        const [platform, connection, build] = url.href.replace('/choose/', '').split('/');
+        const cookie = serialize('session', `${platform},${connection},${build}`, {
+            maxAge: 60 * 60 * 24,
```

Comment:
These magic numbers seem to be commonly used. They appear in the project from time to time. We need to think about moving them to one place and giving them descriptive names.